A32NX/html_ui/Pages/VCockpit/Instruments/NavSystems/LogicElements/A32NX_FlightPlanManager.js

```diff
@@ -1053,6 +1054,91 @@ class FlightPlanManager {
         SimVar.SetSimVarValue("C:fs9gps:FlightPlanNewApproachTransition", "number", transition);
         SimVar.SetSimVarValue("C:fs9gps:FlightPlanLoadApproach", "number", 1);
     }
+    setApproach(approach)  {
+        this._approach = approach;
+    }
+    calculateInitialPattern(turn, distance, course, waypoint) {
+        // console.log("constructing entry pattern for hold at "+ waypoint.ident + "")
+        let courseRad = course * (Math.PI / 180.0)
+        let parallelEntry = [courseRad, ((course + 110) % 360) * (Math.PI / 180.0)];
+        let tearEntry = [courseRad, ((course - 70) % 360) * (Math.PI / 180.0)]
+        let directEntry = [((course+110) % 360) * (Math.PI / 180.0), ((course + 290) % 360) * (Math.PI / 180.0)]
+
+        // let planeHeading = Simplane.getHeadingMagnetic();
+        let headingRad = planeHeading * (Math.PI / 180.0);
+        
+        let entry
+
+        if (headingRad > Math.min(parallelEntry[0], parallelEntry[1]) && headingRad < Math.max(parallelEntry[0], parallelEntry[1])) {
+            entry = "PARALLEL"
+        }
+        if (headingRad > Math.min(tearEntry[0], tearEntry[1]) && headingRad < Math.max(tearEntry[0], tearEntry[1])) {
+            entry = "TEAR"
+        }
+        if (headingRad > Math.min(directEntry[0], directEntry[1]) && headingRad < Math.max(directEntry[0], directEntry[1])) {
+            entry = "DIRECT"
+        }
+
+        let waypoints = [new LatLong(waypoint.infos.lat, waypoint.infos.long)] // start with the hold fix
+
+        let reverseTurnBearing = turn == "L" ? (course + 90) % 360 : (course - 90) % 360;
+        let turnBearing = turn == "R" ? (course + 45) % 360 : (course - 90) % 360;
+        let backCourse = (course + 180) %360
```

Comment:
Those should be `const`

Reply:
Are const arrays still growable/changeable?

Wasn't aware const even exists 👍

Reply:
Const just means the variable cannot be reassigned. Stands for "constant", you can still mutate the array and call various functions upon it.